src/components/MaterialUI/ScopeSelector/TreeView.js

```diff
@@ -3,49 +3,70 @@ import TreeViewMui from "@material-ui/lab/TreeView";
 import TreeItem from "./TreeItem";
 import { scopeTypes } from "./../../../constants";
 
-const TreeView = ({ rootId, getScope, selectedScope, closeSelector }) => {
-	const [expanded, setExpanded] = React.useState(selectedScope.scopePath);
-	const [selected, setSelected] = React.useState(selectedScope.id);
-
-	const virtualScopes = [];
+export const BaseTreeView = ({
+	rootId,
+	getScope,
+	closeSelector,
+	initExpanded,
+	selected,
+	setSelected,
+	redirectOnSelect,
+}) => {
+	const [expanded, setExpanded] = React.useState(initExpanded);
 
 	const handleToggle = (_, nodeIds) => {
 		setExpanded(nodeIds);
 	};
 
-	const handleSelect = (_, nodeId) => {
-		if (virtualScopes.includes(nodeId) === false) {
-			setSelected(nodeId);
-		}
+	const onScopeSelect = (_, nodeId) => {
+		const newSelected = !Array.isArray(selected)
+			? nodeId
+			: selected.includes(nodeId)
+			? selected.filter(id => id !== nodeId)
+			: [nodeId, ...selected];
+
+		setSelected(newSelected);
+		closeSelector(_, newSelected);
 	};
 
 	const renderTree = scopeId => {
 		const currentScope = getScope(scopeId);
 		if (currentScope == null) return null;
 
-		if (currentScope.type === scopeTypes.virtual) {
-			virtualScopes.push(currentScope.id);
-		}
-
```

Comment:
So, since it's removed from there it should be handled somewhere else

Reply:
I didn't find anything what reimplements that behavior. Can you explain please?

Reply:
is Virtual scope check already exist inside TreeItem, why need double check ?